pandas/tests/frame/indexing/test_indexing.py

```diff
@@ -1851,3 +1870,54 @@ def test_setitem_dict_and_set_disallowed_multiindex(self, key):
         )
         with pytest.raises(TypeError, match="as an indexer is not supported"):
             df.loc[key] = 1
+
+
+class TestSetitemValidation:
```

Comment:
is there a viable way to share this with the analogous copy/pasted Series tests?

Reply:
maybe, but I can't think of anything that wouldn't overcomplicate it - any suggestions?

Reply:
Not worth worrying about